docs/tutorial.md

```diff
@@ -1,5 +1,910 @@
+Welcome! If you're new to Hyperapp, you've found the perfect place to start learning.
 
-Under construction. See [Examples](examples.md).
+The Set-up
+-----------------------------------
+
+Together we'll build a simple newsreader-like application. As we do, we'll work
+our way through the five core concepts: view, state, actions, effects and subscriptions.
+
+To move things along, let's imagine we've already made a static version of the
+app we want to build, with this HTML:
+
+
+```html
+<div id="app" class="container">
+    <div class="filter">
+        Filter:
+        <span class="filter-word">ocean</span>
+        <button>&#9998;</button>
+    </div>
+
+    <div class="stories">
+        <ul>
+            <li class="unread">
+                <p class="title">The <em>Ocean </em>is Sinking</p>
+                <p class="author">Kat Stropher</p>
+            </li>
+            <li class="reading">
+                <p class="title"><em>Ocean </em>life is brutal</p>
+                <p class="author">Surphy McBrah</p>
+            </li>
+            <li>
+                <p class="title">
+                    Family friendly fun at the
+                    <em>ocean </em>exhibit
+                </p>
+                <p class="author">Guy Prosales</p>
+            </li>
+        </ul>
+    </div>
+
+    <div class="story">
+        <h1>Ocean life is brutal</h1>
+        <p>
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
+            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+            aliquip ex ea commodo consequat.
+        </p>
+        <p class="signature">Surphy McBrah</p>
+    </div>
+    
+    <div class="autoupdate">
+        Auto update: <input type="checkbox" />
+    </div>
+</div>
+```
+
+...and some CSS [here](https://zaceno.github.com/hatut/style.css).
+
+It looks like this:
+
+![Initial static mockup](./tut1.png)
+
+We'll start by making Hyperapp render the HTML for us. Then we will
+add dynamic behavior to all the widgets, including text input and 
+dynamically fetching stories.
+
+First, let's begin with the traditional "Hello World!"
+
+Hello World
+------------------------------
+
+Create this html file:
+
+```html
+<!doctype html>
+<html>
+  <head>
+    <link rel="stylesheet" href="https://zaceno.github.com/hatut/style.css">
+    <script type="module">
+
+// -- IMPORTS --
+
+import {h, app} from "https://unpkg.com/hyperapp?module"
+
+
+        
+// -- ACTIONS --
+
+
+                
+// -- VIEWS ---
+
+        
+        
+// -- RUN --
+
+app({
+  node: document.getElementById("app"),
+  view: () => h("h1", {}, [
+      "Hello ", 
+      h("i", {}, "World!")
+  ])
+})
+    </script>
+  </head> 
+  <body>
+    <div id="app"></div>
+  </body>
+</html>
+```
+
+> The sections demarked by comments are not important. They're
```

Comment:
I've never heard of `demarked` - but a similar word would be `denoted`. Looks like demarked is a word, just not super common in Canada. Your call.

Reply:
I actually think "demarked" is more appropriate than "denoted" here, because I'm using comments to create a separating boundary between different sections. (https://www.lexico.com/en/definition/demarcate). 

But if it's a strange term, we could perhaps say: "The sections outlined by comment headings ..."?